packages/astro/src/types/public/integrations.ts

```diff
@@ -110,6 +110,13 @@ export interface AstroAdapter {
 	exports?: string[];
 	args?: any;
 	adapterFeatures?: AstroAdapterFeatures;
+	/**
+	 * Determines how the adapter's entrypoint is handled during the build.
+	 * - `'self'`: The adapter defines its own entrypoint and sets rollupOptions.input
+	 * - `'legacy-dynamic'`: Uses the virtual module entrypoint with dynamic exports
+	 * @default 'legacy-dynamic'
+	 */
+	entryType?: 'self' | 'legacy-dynamic';
```

Comment:
Very much inviting bikeshedding of this API change! This is basically what switches to the new entrypoint API.

Reply:
I don't care much about the name tbh, I care most about docs and such. We can add a ticket for that. For example, we'll need to explain what are the implications of `rollupOptions.input`. It seems very internal, so it's worth something more descriptive.